### PR TITLE
Add server setting `aggregate_function_group_array_has_limit_size`

### PR DESCRIPTION
**Pull request by zhongyuankai**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add this `aggregate_function_group_array_has_limit_size`setting to support discarding data in some scenarios.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>Modify your CI run</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  All with Azure
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [x]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

<details>

### CODE REVIEW

**Comment by UnamedRus** [comment]
Already possible via parameter:
```
SELECT length(groupArray(100000)(number))
FROM numbers(100000000)

Query id: 058a99f2-acf6-4c52-ac80-acfacc2d0786

┌─length(groupArray(100000)(number))─┐
│                             100000 │
└────────────────────────────────────┘
```

**Comment by zhongyuankai** [comment]
> Already possible via parameter:
> 
> ```
> SELECT length(groupArray(100000)(number))
> FROM numbers(100000000)
> 
> Query id: 058a99f2-acf6-4c52-ac80-acfacc2d0786
> 
> ┌─length(groupArray(100000)(number))─┐
> │                             100000 │
> └────────────────────────────────────┘
> ```
> 
> But having it as setting will not harm, i guess. ``

@UnamedRus Yes, but it is difficult to change the behavior of a `AggregatedMergeTree` table when it already has data.

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 6098f98cf4ad42631d496446f1bffcfd67e68c09 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63516/6098f98cf4ad42631d496446f1bffcfd67e68c09/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Core/ServerSettings.h [hunk 1]
```diff
@@ -50,6 +50,7 @@ namespace DB
     M(UInt64, max_temporary_data_on_disk_size, 0, "The maximum amount of storage that could be used for external aggregation, joins or sorting., ", 0) \
     M(String, temporary_data_in_cache, "", "Cache disk name for temporary data.", 0) \
     M(UInt64, aggregate_function_group_array_max_element_size, 0xFFFFFF, "Max array element size in bytes for groupArray function. This limit is checked at serialization and help to avoid large state size.", 0) \
+    M(Bool, aggregate_function_group_array_limit_size, false, "This is set to true. Array elements are truncated when they exceed the max array element size.", 0) \
```
Comment (yariks5s) [inline src/Core/ServerSettings.h:53]: Not clear what this setting means. The name of the setting `aggregate_function_group_array_limit_size` tells us that it should contain some kind of size to be limited in groupArray function, but it is a bool. Also, not really clear what means `This is set to true.` when the default setting is set to false😕

**Review by yariks5s — CHANGES_REQUESTED** [review]

File: src/Core/ServerSettings.h (hunk 1 above)
Comment (zhongyuankai) [reply src/Core/ServerSettings.h]: Thank you for your reply, I have renamed and added a new comment.

**Comment by zhongyuankai** [comment]
@yariks5s Failed tests don't seem to be related to changes.

**Review by yariks5s — APPROVED** [review]

**Comment by yariks5s** [comment]
@zhongyuankai can you please merge master? The failed test doesn't seem to be related to your changes.

**Pull request merged**